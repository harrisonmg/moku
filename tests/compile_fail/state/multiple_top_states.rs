use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod state_machine {}

    struct Top;
    impl TopState<BlinkyState> for Top {}

    struct OtherTop;
    impl TopState<BlinkyState> for OtherTop {}
}

fn main() {}
