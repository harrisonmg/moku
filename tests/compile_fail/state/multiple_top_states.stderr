error: multiple `moku::TopState`s are defined within this module
  --> tests/compile_fail/state/multiple_top_states.rs:14:5
   |
14 |     impl TopState<BlinkyState> for OtherTop {}
   |     ^^^^
